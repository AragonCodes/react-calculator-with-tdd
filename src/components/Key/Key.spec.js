import React from 'react';
import { shallow } from 'enzyme';
import Key from './Key';
import renderer from 'react-test-renderer';

describe('Key', () => {
  let wrapper;
  
  beforeEach(() => {
    wrapper = shallow(
      <Key
        keyAction={jest.fn()}
        keyType={''}
        keyValue={''}
      />
    );
  });

  it('should render correctly', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should render a <div />', () => {
    expect(wrapper.find('div').length).toEqual(1);
  });

  it('should render the value of keyValue', () => {
    wrapper.setProps({keyValue: '0'});
    expect(wrapper.text()).toEqual('0');
  });
});