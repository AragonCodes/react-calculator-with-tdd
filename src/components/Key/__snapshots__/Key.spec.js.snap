// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Key should render correctly 1`] = `
"<div className=\\"key-container \\" onClick={[Function: onClick]}>
  <p className=\\"key-value\\" />
</div>"
`;
